src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -242,15 +242,30 @@ pub(crate) fn is_ci_llvm_available(config: &Config, asserts: bool) -> bool {
 
 /// Returns true if we're running in CI with modified LLVM (and thus can't download it)
 pub(crate) fn is_ci_llvm_modified(config: &Config) -> bool {
-    CiEnv::is_rust_lang_managed_ci_job() && config.rust_info.is_managed_git_subrepository() && {
-        // We assume we have access to git, so it's okay to unconditionally pass
-        // `true` here.
-        let llvm_sha = detect_llvm_sha(config, true);
-        let head_sha =
-            output(helpers::git(Some(&config.src)).arg("rev-parse").arg("HEAD").as_command_mut());
-        let head_sha = head_sha.trim();
-        llvm_sha == head_sha
+    // If not running in a CI environment, return false.
+    if !CiEnv::is_ci() {
+        return false;
+    }
+
+    // In rust-lang/rust managed CI, ensure the LLVM submodule is present.
+    if CiEnv::is_rust_lang_managed_ci_job() {
```

Comment:
Is this needed/is this expected to fix anything? We forcefully checkout all submodules at the beginning of each CI job :thinking:

Reply:
Yes, I am aware of that. This is just a way to ensure that it will be available before proceeding with the change check. But now I think it might be better to assert its existence instead.

Reply:
I think that would be clearer, because this was a bit confusing to me ("ensure something that is already ensured somewhere else").

Reply:
I almost wonder if ensuring the LLVM submodule is checked out or is otherwise available should be its own `Step`